### ISSUE TITLE
cursorline, cursorcolumn, colorcolumn broken in new 0.10 colorscheme

### ISSUE THREAD

**Issue by wchargin** [neovim/neovim#29806]
### Problem

The `cursorline`, `cursorcolumn`, and `colorcolumn` settings seem to do nothing in the new default nvim colorscheme:

<img width="505" alt="Screenshot of nvim open in three colorschemes (default, vim, desert), with line and column highlighted in the latter two but not the first" src="https://github.com/user-attachments/assets/9fdcee40-329a-4d5b-90dd-0a24cc824e00">

I quite like the minimalism and contrast of the new scheme overall, but column highlighting is very useful for me for eyeballing whether two pieces of code are at the same indentation level, and line highlighting is useful to draw the eye.

Tested latest release as well as nightly gadef830f8. Using macOS default Terminal.app. Also tested in different Terminal.app color schemes and couldn't find anything that worked.

### Steps to reproduce

```
nvim --clean +':set cursorline cursorcolumn colorcolumn=+1' +':norm 5Oxxxxx' +':norm 3G0ll'
```

Then `:colorscheme vim` and note the difference.

### Expected behavior

The current line is underlined or highlighted in some way, as is the current column.

### Neovim version (nvim -v)

v0.10.0, v0.11.0-dev-470+gadef830f8

### Vim (not Nvim) behaves the same?

no (N/A)

### Operating system/version

macOS Sonoma 14.5

### Terminal name/version

Apple Terminal 2.14 (453)

### $TERM environment variable

xterm-256color

### Installation

GitHub releases, nvim-macos-arm64

**Comment by wchargin** [comment]
Hmm. I had tried `:set termguicolors` but it made things worse (so I understand why it's off by default for Apple Terminal). Is there anything I can configure in Apple Terminal to work around this? From my perspective, this worked fine in nvim 0.9.5, so it feels like a regression, and asking me to switch my whole terminal emulator feels unfortunate…

I just tried downloading iTerm2, and now my neovim background is a strange blue color that both decreases contrast and clashes with the abutting window chrome. Is there a way to customize that? I assume it's intentional but it's surprising to me considering the minimalism of the rest of the colorscheme.

<img width="697" alt="Screenshot of nvim --clean in iTerm2 3.5.3 with default settings" src="https://github.com/user-attachments/assets/9208e5cc-5e2f-4b46-97e5-b82520bbb15a">
